lib/lib/datamanagement/repository/grid_repository.dart

```diff
@@ -7,13 +7,19 @@ class GridRepository {
 
   final ErgastApiClient clientErgast = ErgastApiClient();
 
-  Future<List<Driver>> fetchGrid(int  seasonYear) async {
-    List<Driver> drivers = [];
-    final response = await clientErgast.get(type: GetType.grid, year: seasonYear);
-    if (response.statusCode == 200) {
-      Map<String, dynamic> jsonObject = jsonDecode(response.body);
-      drivers = Driver.listFromJson(jsonObject["MRData"]["DriverTable"]["Drivers"]);
+  Future<Map<String, dynamic>> fetchGrid(int  seasonYear) async {

```

Comment:
A `String: dynamic` típus bűzlik :) nem egyértelmű, hogy ez mi, nagyon könnyű az ilyesmit elrontani programozói oldalon. Itt azzal a feltételezéssel élsz, hogy a repository az API-n kapott választól függően beállít egy map-et, amiben a `data` mezőben fogjuk majd megtalálni az adatot. 

Egyrészt itt is le van duplikálva ez a string többször, másrész mindenhol ahol ezt feldolgozzuk, ott is fen van véve. Gondolj bele mi történik ha ez a string megváltozik/más néven szeretnéd közlekedtetni az adatot: mindenhol módosítanod kell, ezt pedig nagyon könnyű elrejteni. Ezen felül még castolnod és kell az értéket (jelen esetben `List<Driver>`-re) a viewmodelen vagy a service-en ahol használni akarod, ami nem jó. Azt kellene elérni, hogy ne kelljen ilyesmiket csinálni

Ezt a commitot vond vissza vagy refactorold a kódot erről a pontról ahol most vagyunk. A repository rétegben minden függvénynek úgy kéne működni, hogy egy már cast-olt objektumot ad vissza ami a választ képezi le (pl. `Future<List<Driver>> fetchGrid(int  seasonYear) async` ebben az esetben teljesen jó volt) és ha hiba van, exception dobással jelzi a hívónak mi volt a baj. Az exception lehet egyedi típus

```
Rewrite GridRepository and uses with handle exceptions	
cb6d5cc	
Robert Dalnoki <dalnokirobert91@gmail.com>	
2023. Jan 16. 20:39
```